sdks/android/Mentat/library/src/main/java/org/mozilla/mentat/TypedValue.java

```diff
@@ -44,8 +40,7 @@ public TypedValue(Pointer pointer) {
      */
     public Long asLong() {
         if (!this.isConsumed()) {
-            this.value = JNA.INSTANCE.typed_value_into_long(this.rawPointer);
-            this.rawPointer = null;
+            this.value = JNA.INSTANCE.typed_value_into_long(this.consumePointer());
```

Comment:
(off-topic) I realize this isn't part of your changes, but the casting pattern we establish in this class is pretty error-prone. The API lets you do things like `typedValue.asString(); typedValue.asDouble()` and expect something sensible to happen - but I think we'll just crash instead!